### ISSUE TITLE
Addition of `oauth2:` prefix in access token breaks Gitlab deploy tokens in Git clone step

### ISSUE THREAD

**Issue by lanPN85** [PrefectHQ/prefect#10832]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

We are using the Git clone pull step with our self-hosted Gitlab instance, and have been using a [Deploy token](https://docs.gitlab.com/ee/user/project/deploy_tokens/) for the access token for Prefect to clone our code.  This has been working until the most recent Gitlab [update](https://about.gitlab.com/releases/2023/09/18/security-release-gitlab-16-3-4-released/#attacker-can-abuse-scan-execution-policies-to-run-pipelines-as-another-user), which fixed a CVE by preventing deploy tokens from working in an OAuth2 context.

This update, combined with the fact that Prefect automatically prepends `oauth2:` to any access token associated with a Gitlab repository ([code](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/deployments/steps/pull.py#L49)) leads to Gitlab rejecting our deploy token used in conjunction with the prefix, while cloning without the `oauth2:` prefix works fine.

At the moment, we are working around the issue by using a Personal Access Token, however I'd appreciate if there was a way to configure Prefect to not add the prefix so we can switch back to deploy tokens, which are less brittle than personal access tokens.

### Reproduction

```python3
# prefect.yaml excerpt
pull:
  - prefect.deployments.steps.git_clone:
      id: clone-source
      repository: "{{ prefect.variables.source_git_repo }}"
      branch: "{{ prefect.variables.source_git_ref }}"
      access_token: "{{ prefect.blocks.secret.source-repo-token }}"
```

### Error

```python3
remote: HTTP Basic: Access denied. The provided password or token is incorrect or your account has 2FA enabled and you must use a personal access token instead of a password.
```

### Versions

```Text
Version:             2.13.2
API version:         0.8.4
Python version:      3.8.18
Git commit:          dcd8a58f
Built:               Thu, Sep 21, 2023 2:33 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         server
```

**Comment by lanPN85** [comment]
@desertaxle Yes, we are setting the `source-repo-token` secret in the form `<username>:<token>', so the appended form when sent to Gitlab is `oauth2:<username>:<token>`. This form has been working up until the recent update, so we're quite certain it's related to the CVE fix.

**Comment by mianos** [comment]
This is still broken. Did you switch to another git module or start using a personal access token?
